scripts/internal/check_broken_links.py

```
@@ -0,0 +1,119 @@
+  Uses requests module for others are painful to use. REFERENCES[9]
+  Handles redirects, http, https, ftp as well.
+
+REFERENCES:
+Using [1] with some modificatons for including ftp
+[1] http://stackoverflow.com/a/6883094/5163807
+[2] http://stackoverflow.com/a/31952097/5163807
+[3] http://daringfireball.net/2010/07/improved_regex_for_matching_urls
+[4] https://mathiasbynens.be/demo/url-regex
+[5] https://github.com/django/django/blob/master/django/core/validators.py
+[6] https://data.iana.org/TLD/tlds-alpha-by-domain.txt
+[7] https://codereview.stackexchange.com/questions/19663/http-url-validating
+[8] https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD
+[9] http://docs.python-requests.org/
+
+"""
+
+from __future__ import print_function
+
+import os
+import re
+import sys
+import requests
+
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+REGEX = r'(?:http|ftp|https)?://'
+REGEX += r'(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
+
+
+def get_urls(filename):
+    """Extracts all URLs available in specified filename
+    """
+    # fname = os.path.abspath(os.path.join(HERE, filename))
+    # expecting absolute path
+    fname = os.path.abspath(filename)
+    print(fname)
+    text = ''
+    with open(fname) as f:
+        text = f.read()
+
+    urls = re.findall(REGEX, text)
+    # remove duplicates, list for sets are not iterable
+    urls = list(set(urls))
+    # correct urls which are between < and/or >
+    i = 0
+    while i < len(urls):
+        urls[i] = re.sub("[<>]", '', urls[i])
+        i += 1
+
+    return urls
+
+
+def validate_url(url):
+    """Validate the URL by attempting an HTTP connection.
+    Makes an HTTP-HEAD request for each URL.
+    Uses requests module.
+    """
+    try:
+        res = requests.head(url)
+        return res.ok
+    except requests.exceptions.RequestException:
+        return False
+
+
+def main():
+    """Main function
+    """
+    files = sys.argv[1:]
```

also you should return an error if no args are passed:

```
if not files:
     sys.exit("usage: %s <FILES...>" % __name__)
```